src/Scripting/Core/ScriptOptions.cs

```diff
@@ -92,6 +93,16 @@ private static ImmutableArray<MetadataReference> GetDefaultMetadataReferences()
         public ImmutableArray<string> Imports { get; private set; }
 
         /// <summary>
+        /// Specifies whether debugging symbols should be emitted.
+        /// </summary>
+        public bool EmitDebugInformation { get; private set; } = false;
+
+        /// <summary>
+        /// Specifies the encoding to be used when debugging scripts loaded from a file, or that will be saved to a file for debugging purposes.
+        /// </summary>
+        public Encoding FileEncoding { get; private set; } = Encoding.UTF8;
+
```

Comment:
Having default encoding might cause confusion. Even if the file uses UTF8 encoding there are still two kinds of encoding - with BOM and without. If it differs the script won't be debuggable. I'd rather require the user to explicitly specify encoding than to run into these subtleties with the default value.